service/debugger/debugger.go

```diff
@@ -1190,6 +1188,24 @@ func (d *Debugger) Command(command *api.DebuggerCommand, resumeNotify chan struc
 
 	if command.Name != api.SwitchGoroutine && command.Name != api.SwitchThread && command.Name != api.Halt {
 		d.target.ResumeNotify(resumeNotify)
+
+		if clientStatusCh != nil {
+			defer func() {
```

Comment:
I think it's better to move this just before the `return state, err` at the end of the function. I think it's clearer to do it directly instead of using a defer because there is only one path that returns a non nil state and we don't need to actually receive from clientStatusCh in cases where we return an error.

Reply:
Done!